frontend/src/tests/import-export.test.js

```diff
@@ -2,47 +2,226 @@
+     *
+     * @param {[object]} objectData
+     * @param {string} objectType
+     * @param {object} objectSchema
+     * @param {File} file
+     * @returns
+     */
+    function jsonContainsData(file, objectData, objectSchema, objectType) {
+        const correctData = objectData.map((object) =>
+            prepareMinimal[`${objectType}`](object)
+        );
+        const dataJSON = JSON.parse(file.fileBits.toString())[`${objectType}s`];
+        // check each object has correct list of keys
+        dataJSON.forEach((object) =>
+            expect(Object.keys(object)).toEqual(objectSchema.keys)
+        );
+        // check each object has correct value
+        let resultJSON = [];
+        for (const object of dataJSON) {
+            const newObject = {};
+            for (const key of objectSchema.keys) {
+                newObject[key] = object[key];
+            }
+            resultJSON.push(newObject);
+        }
+        expect(resultJSON).toEqual(correctData);
+    }
+
+    /**
+     * Given correct object data, object type, correct object schema, and CSV File object,
+     * carry out tests to check whether the File object contains correct data.
+     *
+     * @param {[object]} objectData
+     * @param {string} objectType
+     * @param {object} objectSchema
+     * @param {File} file
+     * @returns
+     */
+    function csvContainsData(file, objectData, objectSchema, objectType) {
+        const correctData = objectData.map((object) =>
+            prepareMinimal[`${objectType}`](object)
+        );
+        // check each object has correct value
+        const workbook = XLSX.read(file.fileBits[0], { type: "array" });
+        const sheet = workbook.Sheets[workbook.SheetNames[0]];
+        let dataCSV = XLSX.utils.sheet_to_json(sheet, { header: 1 });
+        const keys = dataCSV.shift();
+        // check the keys are correct
+        expect(keys).toEqual(objectSchema.keys);
+        // transform to array of objects
+        dataCSV = dataCSV.map(function (row) {
+            return keys.reduce(function (obj, key, i) {
+                obj[key] = row[i];
+                return obj;
+            }, {});
+        });
+        //check each row has correct value
+        let resultCSV = [];
+        const rowMapper = new SpreadsheetRowMapper(objectSchema);
+        for (const row of dataCSV) {
+            resultCSV.push(rowMapper.formatRow(row));
+        }
+        expect(resultCSV).toEqual(correctData);
+    }
+
+    /**
+     * Given correct object data, object type, and correct object schema,
+     * carry out tests to check whether exporting correct object data to file
+     * will produce correct object data file.
+     *
+     * @param {[object]} objectData
+     * @param {string} objectType
+     * @param {object} objectSchema
+     * @param {string} fileType
+     * @returns
+     */
+    function testExportToFile(objectData, objectType, objectSchema) {
```

Comment:
These functions shouldn't be defined in the `describe` block. They should probably be in their own utils file.